app/api/items/get-navigation-data.go

```diff
@@ -0,0 +1,186 @@
+	StartDate             string  `json:"start_date"` // iso8601 str
+	ValidationDate        string  `json:"validation_date"` // iso8601 str
+	FinishDate            string  `json:"finish_date"` // iso8601 str
+}
+
+type navigationItemAccessRights struct {
+	FullAccess						bool		`json:"full_access"`
+	PartialAccess					bool		`json:"partial_access"`
+	GrayAccess  					bool		`json:"gray_access"`
+}
+
+type navigationItemString struct {
+	// title (from items_strings) in the user’s default language or (if not available) default language of the item
+	Title         				string  `json:"title"`
+}
+
+type navigationItemCommonFields struct {
+	ID                		int64   `json:"id"`
+	Type              		string  `json:"type"`
+	TransparentFolder 		bool	  `json:"transparent_folder"`
+	// whether items.idItemUnlocked is empty
+	HasUnlockedItems  		bool    `json:"has_unlocked_items"`
+
+	String                navigationItemString `json:"string"`
+	User                  navigationItemUser `json:"user"`
+	AccessRights          navigationItemAccessRights `json:"access_rights"`
+
+	Children							[]navigationItemChild `json:"children,omitempty"`
+}
+
+type navigationDataResponse struct {
+	*navigationItemCommonFields
+}
+
+type navigationItemChild struct {
+	*navigationItemCommonFields
+
+	Order 						int64 `json:"order"`
+	AccessRestricted  bool  `json:"access_restricted"`
+}
+
+// Bind .
+func (req *GetItemRequest) Bind(r *http.Request) error {
+	strItemID := chi.URLParam(r, "itemID")
+	itemID, err := strconv.ParseInt(strItemID, 10, 64)
+	if err != nil {
+		return fmt.Errorf("missing itemID")
+	}
+	req.ID = itemID
+	return nil
+}
+
+func (srv *Service) getNavigationData(rw http.ResponseWriter, httpReq *http.Request) service.APIError {
+	req := &GetItemRequest{}
+	if err := req.Bind(httpReq); err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	user := srv.getUser(httpReq)
+	rawData, err := srv.Store.Items().GetRawNavigationData(req.ID, user.UserID, user.DefaultLanguageID())
+	if err != nil {
+		return service.ErrUnexpected(err)
+	}
+
+	accessDetailsMap, err := srv.getAccessDetailsForRawNavigationItems(rawData, user)
+	if err != nil {
+		return service.ErrUnexpected(err)
+	}
+
+	accessDetailsForRootItem, hasAccessDetailsForRootItem := accessDetailsMap[req.ID]
+	if len(*rawData) == 0 || (*rawData)[0].ID != req.ID || !hasAccessDetailsForRootItem ||
+		(!accessDetailsForRootItem.FullAccess && !accessDetailsForRootItem.PartialAccess && !accessDetailsForRootItem.GrayedAccess) {
+		return service.ErrForbidden(errors.New("insufficient access rights on given item id"))
+	}
+
+	response := navigationDataResponse{
+		srv.fillNavigationCommonFieldsWithDBData(&(*rawData)[0], &accessDetailsForRootItem),
+	}
+	idsToResponseData := map[int64]*navigationItemCommonFields{req.ID: response.navigationItemCommonFields}
+	srv.fillNavigationSubtreeWithChildren(rawData, accessDetailsMap, idsToResponseData)
+
+	render.Respond(rw, httpReq, response)
+	return service.NoError
+}
```

Comment:
waw complex all this

Reply:
Yeah, the linter complained about the cyclomatic complexity that's why there are a lot of functions